desktop/src/gui/dialogs/open_dialog.rs

```diff
@@ -517,56 +549,285 @@ impl OpenDialog {
     }
 }
 
-struct OptionalUrlField {
-    value: String,
+trait InnerField {
+    type Value;
+    type Result;
+
+    fn value_if_missing(&self) -> Self::Value;
```

Comment:
Can't this just be default_value? (As far I see it is used for that purpose)

Reply:
I didn't want to confuse it with `default` which OptionalField takes, otherwise in the construction you're providing two "default"s of different types (one `Option<T>`, one `T`)